docs/how-to-build-a-lexical-model.md

```diff
@@ -0,0 +1,365 @@
+      </Languages>
+    </LexicalModel>
+  </LexicalModels>
+</Package>
+```
+
+Step 4.3: Creating the model file
+---------------------------------
+
+Finally, the `model.ts` is a TypeScript source code file that is
+executed to build the model. For our purposes, we can simply copy-paste
+the following example:
+
+```typescript
+import LexicalModelCompiler from "@keymanapp/developer-lexical-model-compiler";
+
+(new LexicalModelCompiler).compile({
+  format: 'trie-1.0',
+  wordBreaking: 'default',
+  sources: ['wordlist.tsv'],
+});
+```
+
+Most importantly, this file specifies that model will use the word list
+in `wordlist.tsv`.
+
+For simple lexical models with exactly one wordlist called
+`wordlist.tsv`, **this file does not need to be changed any further**.
+
+However, it specifies a few other things:
+
+  - `format: 'trie-1.0'`: Use the trie back-end, that allows for
+    prediction based on prefixes of words.
+  - `wordBreaking: 'default'`: use Unicode's default word boundary rules
+    to find where "words" are when predicting text. This works for most
+    languages and writing systems, however, it does not work for
+    languages/writing systems that do not mark breaks between words,
+    e.g., Chinese, Thai, Laos, Khmer, and Japanese, to name a few. For
+    these writing systems, you must provide a language-specific word
+    breaking algorithm to find the words in running text.
+
+Step 5: Build the model
+-----------------------
+
+
+Appendix: Tab-separated wordlist file format specification
+----------------------------------------------------------
+
+<a id="appendix:tsv"></a>
+
+[appendix]: ./#appendix:tsv
+
+The **lexical model compiler** expects wordlists to abide by the
+following **tab-separated values** (TSV) format:
+
+ - the file is must be **UTF-8** encoded text file
+ - newlines are either **LF** or **CRLF**
+ - the file either consists of a **comment** or an **entry**
+ - **comment** lines MUST start with the '#' character on the very first column
+ - **entries** are one or two columns, separated by the (horizontal)
+   **tab character** (U+0009)
+ - column 1 (REQUIRED): the **word form**: can have any character
+   except tab, CR, or LF. Surrounding whitespace characters are trimmed.
+ - column 2 (optional): the **count**: a non-negative integer specifying how many
+   times this entry has appeared in the corpus. Blank means
+   'indeterminate', and is treated as if the word exists in the corpus,
+   but will be predicted at the lowest possible priority.
+ - the following columns are ignored.
+
+> Source:
+> [build-trie.js@e58173f](https://github.com/keymanapp/keyman/blob/307436e7c24caa5c720b272640184362c4dc3223/developer/js/lexical-model-compiler/build-trie.ts#L21-L66)
+
+Exporting a spreadsheet from Google Sheets as a TSV will produce
+properly formatted output. I have not found a reliable output option in
+Microsoft Excel for Mac that will produce output in the expected format.
```

Comment:
Excel for Windows can load the .tsv file okay., but Save has trouble with encoding. However, Copy/Paste into Notepad++ works (older versions of Notepad would work except they prepend a UTF-8 BOM 😭; new versions don't do this).